src/plus/focus/focusProvider.ts

```diff
@@ -313,230 +328,86 @@ export class FocusProvider implements Disposable {
-
-					pinned: enriched?.type === 'pin',
-					snoozed: enriched?.type === 'snooze',
-					sortTime: item.pullRequest.updatedDate.getTime(),
-					repositoryIdentity: {
-						remote: { url: item.pullRequest.refs?.head?.url },
-						name: item.pullRequest.repository.repo,
-						provider: {
-							// TODO: fix this typing, set according to item
-							id: 'github' as GkProviderId,
-							repoDomain: item.pullRequest.repository.owner,
-							repoName: item.pullRequest.repository.repo,
-						},
-					},
-					ref:
-						item.pullRequest.refs?.head != null
-							? {
-									branchName: item.pullRequest.refs.head.branch,
-									sha: item.pullRequest.refs.head.sha,
-									remoteName: item.pullRequest.refs.head.owner,
-							  }
-							: undefined,
-			  }
-			: {
-					type: 'issue',
-					provider: item.issue.provider,
-					id: item.issue.id,
-					uniqueId: item.issue.nodeId!,
+					uuid: providerPr.uuid,
+					provider: pr.pullRequest.provider,
+					enrichable: enrichable,
+					repoIdentity: repoIdentity,
+				};
+			});
+
+			const actionableItems = getActionablePullRequests(
+				inputPrs,
+				{ id: myAccount!.username! },
+				{ enrichedItemsByUniqueId: enrichedItemsByEntityId },
+			);
+			// Map from shared category label to local actionable category, and get suggested actions
+			categorized = actionableItems.map(item => {
+				const actionableCategory = sharedCategoryToFocusActionCategoryMap.get(item.suggestedActionCategory)!;
+				const suggestedActions = prActionsMap.get(actionableCategory)!;
+				return {
+					...item,
 					isNew:
 						this._groupedIds != null &&
-						this._groupedIds.get(item.issue.nodeId!) != focusCategoryToGroupMap.get(category),
-					title: item.issue.title,
-					date: item.issue.updatedDate,
-					author: item.issue.author.name,
-					avatarUrl: item.issue.author.avatarUrl,
-					repoAndOwner: `${item.issue.repository?.owner}/${item.issue.repository?.repo}`,
-					url: item.issue.url,
-
-					enrichable: {
-						type: 'issue',
-						id: item.issue.nodeId!,
-						url: item.issue.url,
-						provider: 'github',
-					},
-					enriched: enriched,
-
-					actionableCategory: category,
-					suggestedActions: [],
-
-					pinned: enriched?.type === 'pin',
-					snoozed: enriched?.type === 'snooze',
-					sortTime: item.issue.updatedDate.getTime(),
-					repositoryIdentity:
-						item.issue.repository != null
-							? {
-									name: item.issue.repository?.repo,
-									provider: {
-										// TODO: fix this typing, set according to item
-										id: 'github' as GkProviderId,
-										repoDomain: item.issue.repository?.owner,
-										repoName: item.issue.repository?.repo,
-									},
-							  }
-							: undefined,
-			  };
+						!this._groupedIds.has(`${item.uuid}:${focusCategoryToGroupMap.get(actionableCategory)}`),
+					actionableCategory: actionableCategory,
+					suggestedActions: suggestedActions,
+				};
+			}) as FocusItem[];
```

Comment:
Can we avoid the `as` here -- maybe `satisfies` or at the type onto the `.map` call?

Reply:
We have to use an `as` somewhere in here because even though the shared provider library function is expecting a certain shape for PRs, we are adding additional fields to the inputs which we expect to see on the output. To prevent TS from thinking we lost those fields, we need to cast until the util function typing can be updated.

For now I will use `satisfies` here and move the `as` up to the output of the `getActionablePullRequests` call to be more accurate with where the issue is.